### PR TITLE
Starting an application ss

### LINKED ISSUE #10: 2. Starting an Application
2. Starting an Application

As a visitor
When I visit the pet index page
Then I see a link to "Start an Application"
When I click this link
Then I am taken to the new application page where I see a form
When I fill in this form with my:
  - Name
  - Street Address
  - City
  - State
  - Zip Code
  - Description of why I would make a good home
And I click submit
Then I am taken to the new application's show page
And I see my Name, address information, and description of why I would make a good home
And I see an indicator that this application is "In Progress"

### LINKED ISSUE #13: 3. Starting an Application, Form not Completed
3. Starting an Application, Form not Completed

As a visitor
When I visit the new application page
And I fail to fill in any of the form fields
And I click submit
Then I am taken back to the new applications page
And I see a message that I must fill in those fields.

### PR DESCRIPTION
**Pull request by stackmm**
This pull request completes User Story 1 and 2 (easier to complete them together).

- Creates a new application form. If the user fully completes the form, then it is stored and they are redirected to that applications show page. If they do not fully complete the form, then they are redirected back to the new application form with an error message to that effect. 
- Also added some test data to the seeds.rb file so we can effectively test the web app on local server. 
- Created a new migration bc I had to modify the "status" attribute of the "applications" table. It needs a default "status" value in order for the new application form to work b/c the new applicant doesn't set the status of their own application - so it needs to be set by default for every new application (set the default value to "In Progress")

### CODE REVIEW

File: spec/features/applications/new_spec.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      expect(page).to have_field("Name")
+      expect(page).to have_field("Street Address")
+      expect(page).to have_field("City")
+      expect(page).to have_field("State")
+      expect(page).to have_field("Zip")
+      expect(page).to have_field("Description")
+      expect(page).to have_button("Submit")
+    end
+
+    it "can fill out form and successfully submit to applicantion's show page" do
```
Comment (kaileykaes) [inline spec/features/applications/new_spec.rb]: spell check on line 19 *application

File: spec/features/applications/new_spec.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      visit "/applications/new"
+
+      fill_in("Name", with: "Kim Jong Un")
```
Comment (kaileykaes) [inline spec/features/applications/new_spec.rb]: Yo yo. I appreciate the irony here, I do. Probs a bit dark if we want to share this with potential employers at the end. Will you pick a less oppressive, factually evil famous person, please?

File: spec/features/pets/index_spec.rb [hunk 3]
```diff
@@ -81,4 +81,13 @@
     expect(page).to have_content(pet_2.name)
```
Comment (kaileykaes) [inline spec/features/pets/index_spec.rb:81]: Is this passing? It seems like you'll need @ before pet_2 and pet_3

File: spec/features/applications/show_spec.rb [hunk 4]
```diff
@@ -76,9 +79,9 @@
       expect(page).to_not have_content("Status:#{@susie.status}")
     end
 
-    it 'applicant show page has link to each pet show page' do 
+    it 'displays a link to each pet show page' do 
       visit "/applications/#{@susie.id}"
-      expect(page).to have_link("#{@pet_1.name}", :href => "/pets/#{@pet_1.id}" )
```
Comment (kaileykaes) [inline spec/features/applications/show_spec.rb:81]: good catch

File: spec/models/application_spec.rb [hunk 5]
```diff
@@ -2,7 +2,17 @@
 
 RSpec.describe Application, type: :model do 
   describe 'relationships' do 
-    it { should have_many :application_pets }
-    it { should have_many(:pets).through(:application_pets) }
+    it {should have_many :application_pets}
+    it {should have_many(:pets).through(:application_pets)}
+  end
+
+  describe 'validations' do
```
Comment (kaileykaes) [inline spec/models/application_spec.rb:9]: I trust you on this, but I don't fully understand what this is solving. Can we follow up on this irl?

File: db/seeds.rb [hunk 6]
```diff
@@ -5,5 +5,56 @@
 #
```
Comment (kaileykaes) [inline db/seeds.rb:5]: Reference point?

File: app/models/application.rb [hunk 7]
```diff
@@ -1,4 +1,13 @@
 class Application < ApplicationRecord
   has_many :application_pets 
   has_many :pets, through: :application_pets
+
+  validates :name, presence: true
```
Comment (kaileykaes) [inline app/models/application.rb:5]: So this is preventing an uncomplete form from being saved?

**Review by kaileykaes — CHANGES_REQUESTED** [review]

**Review by kaileykaes — CHANGES_REQUESTED** [review]
There are a couple of minor changes I requested.

File: app/models/application.rb (hunk 7 above)
Comment (stackmm) [reply app/models/application.rb:5]: that is exactly what it is doing!

File: spec/features/applications/new_spec.rb (hunk 1 above)
Comment (stackmm) [reply spec/features/applications/new_spec.rb]: fixed ty

File: spec/features/pets/index_spec.rb (hunk 3 above)
Comment (stackmm) [reply spec/features/pets/index_spec.rb:81]: This does pass (fyi, this test was all from the original files we forked - I didn't write any of this). If you look at line 72, `pet_2 = Pet.create(adoptable: true, age: 3, breed: "domestic pig", name: "Babe", shelter_id: shelter.id)` ,  pet_2 is created as a simple variable (no @) in the "it" block. Therefore, no @ sign needed when calling it inside of the same "it" block

**Pull request merged**